upcloud/client/client.go

```diff
@@ -142,32 +142,56 @@ func (c *Client) getBaseURL() string {
 
 type configFn func(o *config)
 
+// WithBaseURL modifies the client baseURL
 func WithBaseURL(baseURL string) configFn {
 	return func(c *config) {
 		c.baseURL = baseURL
 	}
 }
 
+// WithInsecureSkipVerify modifies the client's httpClient to skip verifying
+// the server's certificate chain and host name. This should be used only for testing.
+func WithInsecureSkipVerify() configFn {
+	return func(c *config) {
+		if c.httpClient != nil { // #nosec G402 // allow setting InsecureSkipVerify to true as explicitly requested
+			c.httpClient.Transport = &http.Transport{
```

Comment:
Does this now also override other `http.Transport` settings with their default values?

Reply:
Ah, good point. Updated to set either `TLSClientConfig` or `InsecureSkipVerify` instead.